File: 1922-count-good-numbers/1922-count-good-numbers.cpp
class Solution {
public:
    int m=1e9+7;
    long long power(long long a,long long b ){
        if(b==0)return 1;
        long long half=power(a,b/2);
        if(b%2==1)return a*half*half%m;
        return half*half%m;
    }

    int countGoodNumbers(long long n) {
        long long even=power(5,(n+1)/2);
        long long odd=power(4,n/2);
        return even * odd%m;
    }
};

File: 1922-count-good-numbers/1922-count-good-numbers.java
class Solution {
    public
    long m=1000000007;
    public long power(long a,long b){
        if(b==0)return 1;
        long half=power(a,b/2);
        if(b%2==0)return half*half%m;
        return a*half*half%m;
    }
    public int countGoodNumbers(long n) {
        long even=power(5,(n+1)/2);
        long prime=power(4,n/2);
        return (int)(even*prime%m);
    }
}